runtime/src/bank.rs

```diff
@@ -170,6 +170,13 @@ use {
     },
 };
 
+struct StakeReward {
+    pubkey: Pubkey,
+    reward: RewardInfo,
+    balance: u64,
+    account: AccountSharedData,
+}
```

Comment:
I think you can move this struct inside the `pay_validator_rewards_with_thread_pool` function; It is not used anywhere else.

`pubkey` is ambitious. Please rename to `stake_pubkey`.
Same for `account -> stake_account`.

`balance -> stakers_reward`